### ISSUE TITLE
A sentence with translation appears in search results for sentences that have no translations

### ISSUE THREAD

**Issue by trang** [Tatoeba/tatoeba2#2596]
**To Reproduce**

* Go to https://tatoeba.org/eng/sentences/search?query=Tom+enjoys+listening&from=und&to=und&user=&orphans=no&unapproved=no&has_audio=&tags=&list=&native=&trans_filter=exclude&trans_to=und&trans_link=direct&trans_user=&trans_orphan=&trans_unapproved=&trans_has_audio=&sort=relevance&sort_reverse=
* The sentence `Tom enjoys listening to heavy metal.` appears in the results.

**Expected behavior**

The search is supposed to display sentences that do not have translations so `Tom enjoys listening to heavy metal.` should not be listed in there.

**Additional information**

Reported in a comment on a the sentence:
https://tatoeba.org/eng/sentences/show/8573851#comment-1229295

I suppose this happens because the sentence was not reindexed after it was merged by Horus.

**Comment by AndiPersti** [comment]
> I suppose this happens because the sentence was not reindexed after it was merged by Horus.

Yes, I've just checked on dev that Horus doesn't update "reindex_flags" when merging links.
